From 25c3dc15b5faacf447cced0b52179344888d0c98 Mon Sep 17 00:00:00 2001
From: Levin Calado <levincalado@gmail.com>
Date: Wed, 14 Oct 2015 23:08:31 +0800
Subject: [PATCH 0877/1168] Revert "fib_rules: fix fib rule dumps across
 multiple skbs"

This reverts commit d0550a3f2d313a5310ace03299d45ef63edfb750.

Signed-off-by: Stefan Guendhoer <stefan@guendhoer.com>

Former-commit-id: b9c08df3a1a483867f6a2d6c4d1e2825dc885fe0
---
 net/core/fib_rules.c | 14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)

diff --git a/net/core/fib_rules.c b/net/core/fib_rules.c
index 1b10e36..da78f5c 100644
--- a/net/core/fib_rules.c
+++ b/net/core/fib_rules.c
@@ -645,17 +645,15 @@ static int dump_rules(struct sk_buff *skb, struct netlink_callback *cb,
 {
 	int idx = 0;
 	struct fib_rule *rule;
-	int err = 0;
 
 	rcu_read_lock();
 	list_for_each_entry_rcu(rule, &ops->rules_list, list) {
 		if (idx < cb->args[1])
 			goto skip;
 
-		err = fib_nl_fill_rule(skb, rule, NETLINK_CB(cb->skb).portid,
-				       cb->nlh->nlmsg_seq, RTM_NEWRULE,
-				       NLM_F_MULTI, ops);
-		if (err)
+		if (fib_nl_fill_rule(skb, rule, NETLINK_CB(cb->skb).portid,
+				     cb->nlh->nlmsg_seq, RTM_NEWRULE,
+				     NLM_F_MULTI, ops) < 0)
 			break;
 skip:
 		idx++;
@@ -664,7 +662,7 @@ skip:
 	cb->args[1] = idx;
 	rules_ops_put(ops);
 
-	return err;
+	return skb->len;
 }
 
 static int fib_nl_dumprule(struct sk_buff *skb, struct netlink_callback *cb)
@@ -680,9 +678,7 @@ static int fib_nl_dumprule(struct sk_buff *skb, struct netlink_callback *cb)
 		if (ops == NULL)
 			return -EAFNOSUPPORT;
 
-		dump_rules(skb, cb, ops);
-
-		return skb->len;
+		return dump_rules(skb, cb, ops);
 	}
 
 	rcu_read_lock();
-- 
2.7.4

